posthog/hogql/hogql.py

```diff
@@ -12,11 +14,18 @@ def translate_hogql(query: str, context: HogQLContext, dialect: Literal["hogql",
 
     try:
         if context.select_team_id:
+            # Only parse full SELECT statements if we have a team_id in the context.
```

Comment:
I'm a bit confused why this parameter determines what happens here.

Come back here after I check how/where its populated & usage.

Reply:
You're right, this could have been greatly simplified.

This is the weird world of interoperability between full HogQL queries and HogQL fragments in property expressions in existing insights.

This `translate_hogql` function is only called from within existing "non-hogql-based" queries. So all trends, session recording lists, etc that use HogQL filters end up here.

<img width="770" alt="image" src="https://user-images.githubusercontent.com/53387/220366341-de55217a-2de7-4142-87cb-06db55efc2d8.png">

Plus also from `test_printer`. That's the only place that was calling this function to print full select queries, and triggering this `if`. I removed all of that, and made the test test things differently.

--

I think `HogQLContext` needs a full refactor. It's from the time of the Python AST printer. I'd store most of its values on the `Printer` class directly. Since this is a whole other set of work with about 50 changed files likely by the end of it, I haven't taken it on yet.

If we do actually move all queries over to HogQL as their base, we can likely keep the `HogQLContext` working as it's working now until we refactor it all out...